account_payment_return/models/payment_return.py

```diff
@@ -108,6 +108,13 @@ def _prepare_invoice_returned_cancel_vals(self):
         return {'returned_payment': False}
 
     @api.multi
+    def unlink(self):
+        if self.filtered(lambda x: x.state == 'done'):
+            raise UserError(_(
+                "You can not remove a payment return if state is 'Done'"))
+        return super(PaymentReturn, self).unlink()
```

Comment:
OK, thanks for clarifying. Odoo doesn't do this kind of things by security rules, so there should be a reason for that. Maybe performance as pointed by Carlos, or maybe to not bypass this check removing access rule.